js/linked-list.js

```diff
@@ -0,0 +1,99 @@
+"use strict";
+
+class Node {
+    constructor(data, next=null){
+        this.data = data;
+        this.next = next;
+    }
+}
+
+class LinkedList {
+    constructor(iterable=null) {
+        this.head = null;
+        this._counter = 0;
+        if(Array.isArray(iterable)){
+            iterable.map(x => this.push(x));
+        }
+    }
+
+    push(val){
+        let newHead = new Node(val, this.head);
+        this.head = newHead;
+        this._counter ++;
+    }
+
+    pop(){
+        if(this.head === null){
+            throw new Error('IndexError: Nothing to pop.');
+        }
+        let output = this.head;
+        this.head = this.head.next;
+        this._counter --;
+        return output.data;
+    }
+
+    search(val){
+        let searchThrough = this.head;
+        try{
+            while(this.head){
+                if(val === searchThrough.data){
+                    return searchThrough.data;
+                }else{
+                    searchThrough = searchThrough.next;
+                }
+            }
+        }catch(e){
+            return 'No such Node';
+        }
+    }
```

Comment:
This method is a little broken because your `while` condition is looking at whether or not `this.head` always has a value, not if your iteration forward through the linked list is actually yielding a new node to you